### DISCUSSION TITLE
Failed to Update ?

### DISCUSSION THREAD

**Discussion by DanMarquesPT** [Romanitho/Winget-AutoUpdate#453] [General]
Hi all,

Having a few issues and kind of lost.

My pc shows latest version installed. Logs say: 
21:08:07 - WAU current version: 1.17.9-4

Having an issue when WAY trying to update to latest, keeps saying old version installed and PowerShell window pops up for half a second

**Comment by Romanitho** [comment]
try uninstall and reinstall.

**Comment by DanMarquesPT** [comment]
I did that, deleted the folder under ProgramData and still says older version compared to the real version

**Reply by Romanitho** [reply]
deleted folder is not enough. You need to run the uninstaller to clean registry

**Reply by DanMarquesPT** [reply]
Same result.

```
########################################################
#     CHECK FOR APP UPDATES (System context with ServiceUI) - 28/10/2023
########################################################
17:16:13 - WAU Policies management Enabled.
17:16:13 - Running Post Update actions:
17:16:13 - -> Creating SymLink for log file (WAU-updates) in Intune Management Extension log folder
17:16:13 - -> Checking if WinGet is installed/up to date
17:16:14 - -> WinGet is up to date: v1.6.2771
17:16:14 - -> Checking Mods Directory:
17:16:14 - -> The mods directory was already secured!
17:16:14 - -> C:\ProgramData\Winget-AutoUpdate\Version.txt removed.
17:16:20 - Post Update actions finished
17:16:20 - Notification Level: None. Notification Language: English
17:16:20 - Checking internet connection...
17:16:20 - Connected !
17:16:21 - Winget Version: v1.6.2771
17:16:21 - WAU current version: 1.17.9-4
17:16:21 - WAU AutoUpdate is Enabled.
17:16:21 - WAU Available version: 1.18.0
17:16:21 - Downloading the GitHub Repository version 1.18.0
17:16:22 - Unzipping the WAU Update package
17:16:23 - Updating WAU...
17:16:23 - Done. Cleaning temp files...
17:16:23 - WAU Update completed.
17:16:23 - Re-run WAU
 
########################################################
#     CHECK FOR APP UPDATES (System context with ServiceUI) - 28/10/2023
########################################################
17:16:24 - WAU Policies management Enabled.
17:16:24 - Running Post Update actions:
17:16:24 - -> Creating SymLink for log file (WAU-updates) in Intune Management Extension log folder
17:16:24 - -> Checking if WinGet is installed/up to date
17:16:24 - -> WinGet is up to date: v1.6.2771
17:16:24 - -> Checking Mods Directory:
17:16:24 - -> The mods directory was already secured!
17:16:24 - -> C:\ProgramData\Winget-AutoUpdate\Version.txt removed.
17:16:29 - Post Update actions finished
17:16:29 - Notification Level: None. Notification Language: English
17:16:29 - Checking internet connection...
17:16:29 - Connected !
```

**Comment by KnifMelti** [comment]
Run the uninstall, manually delete `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate` then install the latest 1.18.0...

**Reply by Romanitho** [reply]
I agree. You have installed a pre-release version that was probably buggy.
Clean as mentioned

**Reply by DanMarquesPT** [reply]
Still the same.
Just tried and the Uninstall process deletes the Registry entry. I guess it's fine because in Installed Apps says latest version. But can't find the place where that information is kept.

**Reply by Romanitho** [reply]
Have a look in here:
`HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate`
And delete if you have any.

One prerelease version of WAU was buggy and wrote the version there. Maybe you've installed the buggy one.

**Reply by DanMarquesPT** [reply]
You are 100% correct :)

Thanks

**Reply by Romanitho** [reply]
no worries. Thanks to you to report and support the project ;)

**Reply by Romanitho** [reply]
oh, and in corporate environment I would suggest to use only "stable" releases 😇😇

**Reply by DanMarquesPT** [reply]
That was my big mistake. I have been testing this on my machine with GPO policy through Intune. When i've decided to push this to all forgot to edit the policy and released the Nightly build.

**Reply by Romanitho** [reply]
😅

**Reply by DanMarquesPT** [reply]
Will try and solve this with a GPO to delete that Entry :)

**Reply by DanMarquesPT** [reply]
For someone with the same issue, needs to delete these entries 
HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate

![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/128723180/9c564179-107d-4f46-81bd-af7c844f0f9a)

**Reply by Romanitho** [reply]
yes. Clean `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate`, your GPO will reconfigure that properly.
